cpp/models/graph_abm/graph_abmodel.h

```diff
@@ -0,0 +1,213 @@
+/*
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Julia Bicker
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef MIO_ABM_GRAPH_ABMODEL_H
+#define MIO_ABM_GRAPH_ABMODEL_H
+
+#include "abm/location_type.h"
+#include "abm/model.h"
+#include "abm/person_id.h"
+#include "abm/time.h"
+#include "abm/location_id.h"
+#include "memilio/utils/compiler_diagnostics.h"
+#include "memilio/utils/logging.h"
+#include "memilio/utils/mioomp.h"
+#include "abm/mobility_rules.h"
+#include "abm/mobility_rules.h"
+#include <cstddef>
+#include <cstdint>
+#include <list>
+#include <vector>
+
+namespace mio
+{
+using namespace abm;
+class GraphABModel : public abm::Model
+{
+    using Base = Model;
+
+public:
+    GraphABModel(size_t num_agegroups, int id,
+                 std::vector<Base::MobilityRuleType> mobility_rules =
+                     std::vector<Base::MobilityRuleType>{&get_buried, &return_home_when_recovered, &go_to_hospital,
+                                                         &go_to_icu, &go_to_school, &go_to_work, &go_to_shop,
+                                                         &go_to_event, &go_to_quarantine})
+        : Base(num_agegroups, id)
+    {
+        Base::m_mobility_rules = mobility_rules;
+    }
+
+    /**
+     * @brief Get person buffer. 
+     */
+    std::vector<size_t>& get_person_buffer()
+    {
+        return m_person_buffer;
+    }
+
+    /** 
+    * @brief Removes person from the model.
+    * @param[in] pos Index of person in m_persons vector.
+    */
+    void remove_person(size_t pos)
+    {
+        Base::m_persons.erase(Base::m_persons.begin() + pos);
+        Base::m_activeness_statuses.erase(Base::m_activeness_statuses.begin() + pos);
+        if (Base::m_person_ids_equal_index) {
+            Base::m_person_ids_equal_index = false;
+        }
```

Comment:
This if clause should be unnecessary, as the result will always be false. (So just keep line 74.)